bin/hdfcoinc/pycbc_multiifo_combine_statmap

```diff
@@ -0,0 +1,220 @@
+f.attrs['timeslide_interval'] = files[0].attrs['timeslide_interval']
+
+f.attrs['background_time'] = files[0].attrs['background_time']
+f.attrs['foreground_time'] = files[0].attrs['foreground_time']
+f.attrs['background_time_exc'] = files[0].attrs['background_time_exc']
+f.attrs['foreground_time_exc'] = files[0].attrs['foreground_time_exc']
+
+
+# combine times that are foreground vetoed as they may differ between bins
+
+for fi in files:
+    ifos_in_file = [ifo for ifo in args.ifos if ifo in fi['segments']]
+    ifo_to_get = 'segments/{}'.format(ifos_in_file[0])
+    group_to_save = 'segments/{}'.format(''.join(ifos_in_file))
+    f[group_to_save + '/end'] = fi[ifo_to_get + '/end'][:]
+    f[group_to_save + '/start'] = fi[ifo_to_get + '/start'][:]
+
+key_set = name_all_datasets(files)
+# copy and concatenate all the columns in the foreground group 
+# from all files /except/ the IFO groups, we recalculate ifar/fap later
+for key in key_set:
+    if key.startswith('foreground') and not any([ifo in key for ifo in args.ifos]):
+        pycbc.io.combine_and_copy(f, files, key)
+
+
+#for the ifo-specific groups, copy them over into the new file
+# - if the ifo is not present for each coincident trigger, 
+#  indicate by including a -1 in the combined file
+
+trigger_list = []
+all_trig_times = {}
+all_trig_ids = {}
+for ifo in args.ifos:
+    all_trig_times[ifo] = numpy.array([], dtype=numpy.uint32)
+    all_trig_ids[ifo] = numpy.array([], dtype=numpy.uint32)
+    # for each file, append the trigger time and id data for each ifo
+    #   - if it isnt there, then fill with -1 values
+for f_in in files:
+    for ifo in args.ifos:
+        if ifo in ifo in f_in['foreground']:
+            all_trig_times[ifo] = numpy.concatenate([all_trig_times[ifo], \
+                                     f_in['foreground/{}/time'.format(ifo)][:]])
+            all_trig_ids[ifo] = numpy.concatenate([all_trig_ids[ifo], 
+                                     f_in['foreground/{}/trigger_id'.format(ifo)][:]])
+        else:
+            all_trig_times[ifo] = numpy.concatenate([all_trig_times[ifo],
+                                     -1*numpy.ones_like(f_in['foreground/fap'][:], dtype=numpy.uint32)])
+            all_trig_ids[ifo] = numpy.concatenate([all_trig_ids[ifo],
+                                     -1*numpy.ones_like(f_in['foreground/fap'][:], dtype=numpy.uint32)])
+
+for ifo in args.ifos:
+    f['foreground/{}/time'.format(ifo)] = all_trig_times[ifo]
+    f['foreground/{}/trigger_id'.format(ifo)] = all_trig_ids[ifo]
+
+#copy across ifar and ifar_exc, etc 
+# recalculate ifar/fap for the foreground triggers by 
+#  applying a trials factor = num_files(num_bins)
+
+all_times = ()
+for ifo in args.ifos:
+    all_times = all_times +  (f['foreground/%s/time' % ifo][:],)
+
+trials_factors = []
+for tc in zip(*all_times):
+    n_coinc = 0
+    test_time = pycbc.events.mean_if_greater_than_zero(tc)
+    for key in f['segments']:
```

Comment:
I think this block could be simplified by using existing segments functions from `pycbc/events/veto.py `. 
Also have a comment before the for loop to say briefly what it's doing

Reply:
Done